lib/pf/web/dispatcher.pm

```diff
@@ -53,205 +52,184 @@ Reference: http://perl.apache.org/docs/2.0/user/handlers/http.html#PerlTransHand
-            my $parsed = APR::URI->parse($r->pool,$r->headers_in->{'Referer'});
-            if ($s->port eq '80' && $proto eq 'https' && $r->uri !~ /$WEB::ALLOWED_RESOURCES/o && $parsed->path !~ /$WEB::ALLOWED_RESOURCES/o) {
-                #Generate a page with a refresh tag
-                $r->handler('modperl');
-                $r->set_handlers( PerlResponseHandler => \&html_redirect );
-                return Apache2::Const::OK;
-            } else {
-                # DECLINED tells Apache to continue further mod_rewrite / alias processing
-                return Apache2::Const::DECLINED;
-            }
-        }
-        else {
-            # DECLINED tells Apache to continue further mod_rewrite / alias processing
-            return Apache2::Const::DECLINED;
-        }
-    }
-    if ($r->uri =~ /$WEB::ALLOWED_RESOURCES_MOD_PERL/o) {
+
+    # Captive-portal resource | RELEASE
+    if ( $uri =~ /$WEB::URL_RELEASE/o ) {
+        $logger->info("URI '$uri' (URL: $url) is a RELEASE request");
         $r->handler('modperl');
-        if ($r->uri =~ /$WEB::MOD_PERL_WISPR/o) {
-            $r->pnotes->{session_id} = $1;
-            $r->set_handlers( PerlResponseHandler => ['pf::web::wispr'] );
-        }
+        $r->set_handlers( PerlResponseHandler => ['pf::web::release'] );
         return Apache2::Const::OK;
     }
 
-    # fallback to a redirection: inject local redirection handler
-    $r->handler('modperl');
-    $r->set_handlers( PerlResponseHandler => \&redirect );
-    # OK tells Apache to stop further mod_rewrite / alias processing
-    return Apache2::Const::OK;
-}
+    # Captive-portal resource | WISPr
+    if ( $uri =~ /$WEB::URL_WISPR/o ) {
+        $logger->info("URI '$uri' (URL: $url) is a WISPr request");
+        $r->handler('modperl');
+        $r->set_handlers( PerlResponseHandler => ['pf::web::wispr'] );
+        return Apache2::Const::OK;
+    }
 
-=item redirect
+    # Captive-portal resources
+    # We don't want to continue in the dispatcher if the requested URI is supposed to reach the captive-portal (Catalyst)
+    # - Captive-portal itself
+    # - Violation pages
+    # - Portal profile filters are handled by Catalyst
+    # See L<pf::web::constants::CAPTIVE_PORTAL_RESOURCES>
+    if ( $uri =~ /$WEB::CAPTIVE_PORTAL_RESOURCES/o ) {
+        $logger->info("URI '$uri' (URL: $url) is properly handled and should now continue to the captive-portal / Catalyst");
+        return Apache2::Const::DECLINED;
+    }
 
-For simplicity and performance this doesn't consume and leverage
-L<pf::Portal::Session>.
+    # Portal-profile filters
+    # TODO: Migrate to Catalyst
+    if ( defined($WEB::ALLOWED_RESOURCES_PROFILE_FILTER) && $uri =~ /$WEB::ALLOWED_RESOURCES_PROFILE_FILTER/o ) {
+        $logger->info("Matched profile uri filter for $uri");
+        #Send the current URI to catalyst with the pnotes
+        $r->pnotes(last_uri => $uri);
+        return Apache2::Const::DECLINED;
+    }
 
-=cut
+    # Proxy passthrough
```

Comment:
From my point of view, captive-portal resources are the most request we may receive (outside of requests that needs to be redirected...) but I can move it upper since it is only a simple if regex and doesn't involve any instanciation or whatnot that may require useless processing